Pinpoint.Win/Views/SettingsWindow.xaml.cs

```diff
@@ -151,11 +153,38 @@ public void PluginChange_Added(object sender, IPlugin plugin, object target)
 
         public void PluginChange_Removed(object sender, IPlugin plugin, object target) => Model.Plugins.Remove(plugin);
 
+        private void CheckboxLaunchOnStartupChecked(object sender, RoutedEventArgs evt)
+        {
+            ToggleRunOnStartupSetting(true);
+        }
+
+        private void CheckboxLaunchOnStartupUnchecked(object sender, RoutedEventArgs evt)
+        {
+            ToggleRunOnStartupSetting(false);
+        }
+
         private void BtnReCenterWindow_OnClick(object sender, RoutedEventArgs e)
         {
             App.Current.MainWindow.MoveWindowToDefaultPosition();
             Close();
             App.Current.MainWindow.Show();
         }
+
+        private void ToggleRunOnStartupSetting(bool shouldRunOnStartup)
+        {
+            var path = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Run";
+
+            var registryKey = Registry.CurrentUser.OpenSubKey(path, true);
+            var applicationExecutablePath = Process.GetCurrentProcess().MainModule.FileName;
+
+            if (shouldRunOnStartup)
+            {
+                registryKey.SetValue("Pinpoint", applicationExecutablePath);
```

Comment:
Does this require Pinpoint to be run with elevated privileges?

Reply:
If so, we should probably consider warning the user if they haven't done so to prevent confusion when it doesn't work

Reply:
No this should work without running with elevated privileges. However, it will only run on startup for the current user, which I think would be the expected behavior.